ceph.spec.in

```diff
@@ -179,6 +179,7 @@ BuildRequires:	xfsprogs
 BuildRequires:	xfsprogs-devel
 BuildRequires:	xmlstarlet
 BuildRequires:	yasm
+BuildRequires:	nasm
```

Comment:
> install-deps.sh: install nasm for isa-l library nasm support build isa-l:AVX512 algorithm implementation while yasm doens't support it. Install nasm assembler to build isa-l

citation needed. i found https://github.com/yasm/yasm/issues/101 .

if nasm is more capable in the sense of instructionset support. probably we can switch over to nasm instead of installing both of them?

Reply:
@tchaikov 
Yes. Below commit's message shows that why we still keep yasm here:
https://github.com/ceph/ceph/pull/37751/commits/8670c0f7b57b3d8bb0dfef6f2cb54950907c6eb7

Reply:
again, please add the citation in the commit message. also, i'd like to avoid installing two assemblers.

> After using isa-l's crc32 implementation to replace crc32c_intel_fast_asm, we'll remove yasm directly.

could you include this change in this patchset?

Reply:
@tchaikov 
I'll follow your suggestion to add the citation.

Could we file another PR to use isa-l's crc32 implementation to replace crc32c_intel_fast_asm? This PR's target is to add nasm to support build isa-l  & isa-l_crypto library.
Do you think that we need to keep only one assembler after merging this PR?

Reply:
i understand the goal of this PR. but i am not sure if this change is really urgent. and i still don't want to install two assemblers. two reasons

1. slow down the build
1. more complexity.

Reply:
@tchaikov 
OK, I'll try to use isa-l's crc32 and remove yasm assembler in this RP.

Reply:
@tchaikov 
There are still something as below that can't be solved by nasm. I'll refine crc32 here after getting reply from internal team working on nasm development.
https://github.com/ceph/ceph/pull/37751/commits/60b68d7bfeb4804dfacbe5ea15e881307547f1a1#diff-d95dccd42b204605642df482b2c25755b8c8d126f58ba89ca0322895fdcafe26R202

You can review other changes in this PR when you're avaiable.

Reply:
@tchaikov 
finish integrating isa-l's crc32 into Ceph. yasm has been removed in this PR.